lib/trino-parquet/src/main/java/io/trino/parquet/reader/ColumnReaderFactory.java

```diff
@@ -132,6 +137,18 @@ public static ColumnReader create(PrimitiveField field, DateTimeZone timeZone, b
             if (DOUBLE.equals(type) && primitiveType == PrimitiveTypeName.DOUBLE) {
                 return new FlatColumnReader<>(field, ValueDecoders::getDoubleDecoder, LONG_ADAPTER);
             }
+            if (type instanceof TimestampType timestampType && primitiveType == INT96) {
```

Comment:
Are there plans to replace this giant wall of ifs with something nicer?

Reply:
It's similar approach as `ParquetWriters#getValueWriter` but more complicated as the reader has to deal with more possibilities than the writer.
This approach was preferred during code review over the shorter (but harder to reason about) approach of dispatching over parquet primitive type first and then matching the trino type and parquet annotations in some order.